packages/scenes-app/src/demos/behaviors/VariableEffectBehavior.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  SceneObjectBase,
+  SceneObjectState,
+  SceneVariable,
+  SceneVariableState,
+  VariableDependencyConfig,
+} from '@grafana/scenes';
+
+interface VariableEffectBehaviorState extends SceneObjectState {
+  // The names of the variables to effect when they change
+  variables: string[];
+  // The effect to run when a variable changes. Optionally return a cancel function
+  effect: (variableState: SceneVariableState) => (() => void) | void;
+}
```

Comment:
```suggestion
interface VariableChangedActionBehaviorState extends SceneObjectState {
  // The names of the variable to subscribe to changes to 
  variableName: string;
  // The handler to run when a variable changes. Return a cancellation function if you do anything async like issue a query
  onChange: (variable: SceneVariable) => (() => void) | void;
} 
```

Wonder if it makes more sense to limit this to a single variable, and the effect handler to take the full SceneVariable instance? That way it could potentially be made generic as well (ie typed SceneVariable). 

The array of variables becomes a bit strange behavior-wise, if multiple variables are specified, will the effect trigger for each if many are changed in one variable change transaction?

Reply:
Yes, I think this is better to handle a single variable effect!